core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/math/percentile.kt

```diff
@@ -0,0 +1,138 @@
+package org.jetbrains.kotlinx.dataframe.math
+
+import org.jetbrains.kotlinx.dataframe.impl.asList
+import java.math.BigDecimal
+import java.math.BigInteger
+import kotlin.reflect.KType
+import kotlin.reflect.typeOf
+
+public inline fun <reified T : Comparable<T>> Iterable<T>.percentileOrNull(percentile: Double): T? =
+    percentile(percentile, typeOf<T>())
+
+public inline fun <reified T : Comparable<T>> Iterable<T>.percentile(percentile: Double): T =
+    percentileOrNull(percentile)!!
+
+public inline fun <reified T : Comparable<T>> Iterable<T>.q1OrNull(): T? = q1(typeOf<T>())
+
+public inline fun <reified T : Comparable<T>> Iterable<T>.q1(): T = q1OrNull()!!
+
+public inline fun <reified T : Comparable<T>> Iterable<T>.q3OrNull(): T? = q3(typeOf<T>())
+
+public inline fun <reified T : Comparable<T>> Iterable<T>.q3(): T = q3OrNull()!!
+
+@PublishedApi
+internal inline fun <reified T : Comparable<T>> Iterable<T?>.q1(type: KType): T? = percentile(25.0, type)
+
+@PublishedApi
+internal inline fun <reified T : Comparable<T>> Iterable<T?>.q3(type: KType): T? = percentile(75.0, type)
+
+@PublishedApi
+internal inline fun <reified T : Comparable<T>> Iterable<T?>.percentile(percentile: Double, type: KType): T? {
```

Comment:
It's fine for now to return the same type it receives, as that's what median did too. However, I'll likely change it pretty soon according to this table: https://github.com/Kotlin/dataframe/issues/961

Reply:
Yes, we could reconsider this logic, I add all of this in accordance with  existing  `median`